app/src/main/res/layout/weather_data_fragment.xml

```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+
+<layout>
+
+    <data>
+
+        <variable
+            name="weatherData"
+            type="com.example.mytestproject.ui.models.weatherDataModel.WeatherData" />
+
+    </data>
+
+    <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto"
+        xmlns:tools="http://schemas.android.com/tools"
+        android:id="@+id/fragment_1_background"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:background="@drawable/gradient_list"
+        tools:context=".ui.weatherData.WeatherDataFragment">
+
+
+        <TextView
+            android:id="@+id/textLocation"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_centerHorizontal="true"
+            android:layout_marginTop="50dp"
+            android:text="@{weatherData.city_name}"
+            android:textColor="@color/text_color"
+            android:textSize="36sp"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <TextView
+            android:id="@+id/textTemperature"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="30dp"
+            android:text="@{Double.toString(weatherData.data.get(0).temp)}"
```

Comment:
Too complicated logic for binding. 
WeatherData class should contain only the data that we display on the screen: 
cityName: String
temperature: String
iconUrl: String

Let's remove list, and change iconId to full iconUrl. Mapper (API -> UI) should do all this logic.